exonum/src/runtime/mod.rs

```diff
@@ -12,109 +12,110 @@
 //!   in order to continue functioning.
-//!   If a node has not deployed the artifact previously, deployment becomes blocking; the node
+//!   If a node has not deployed the artifact previously, deployment becomes blocking. The node
 //!   does not participate in consensus or block processing until the deployment is completed
 //!   successfully. If the deployment is unsuccessful, the node stops indefinitely.
-//!   Due to deployment confirmation mechanics built into the supervisor, it is reasonable
+//!   The deployment confirmation mechanics is built into the supervisor. Thus, it is reasonable
 //!   to assume that a deployment failure at this stage is local to the node and
 //!   could be fixed by the node admin.
 //!
 //! # Service Lifecycle
 //!
-//! 1. Once the artifact is committed, it is possible to instantiate a corresponding service.
-//!   Each instantiation request contains an ID of a previously deployed artifact,
+//! 1. Once the artifact is committed, it is possible to instantiate the corresponding service.
+//!   Each instantiation request contains an ID of the previously deployed artifact,
 //!   a string instance ID, and instantiation arguments in a binary encoding
-//!   (by convention, Protobuf). As with the artifacts, the logic controlling instantiation
+//!   (by convention, Protobuf). As with the artifacts, the logic that controls instantiation
 //!   is encapsulated in the supervisor service.
 //!
-//! 2. During instantiation, the service is assigned a numeric ID, which is used to reference
+//! 2. During instantiation the service gets a numeric ID, which is used to reference
 //!   the service in transactions. The runtime can execute initialization logic defined
 //!   in the service artifact; e.g., the service may store some initial data in the storage,
-//!   check service dependencies, etc. The service (or the enclosing runtime) may signal that
-//!   the initialization failed, in which case the service is considered not instantiated.
+//!   check service dependencies, etc. If the service (or the enclosing runtime) signals that
+//!   the initialization failed, the service is considered not instantiated.
 //!
 //! 3. Once the service is instantiated, it can process transactions and interact with the
 //!   external users in other ways. Different services instantiated from the same artifact
 //!   are independent and have separate blockchain storages. Users can distinguish services
 //!   by their IDs; both numeric and string IDs are unique within a blockchain.
 //!
-//! Warning: state hash of the block is not affected by services, instantiated within it. The
-//! only exception is the genesis block where object hashes of builtin services are included
-//! to the resulting block state hash. This behavior difference will be unified in next release.
+//! Warning: the services, instantiated within a block, do not affect the state hash of that block. The
```

Comment:
This doesn't seem true any more with the automated state hash aggregation. @alekseysidorov, could you confirm? If so, the paragraph should be removed.

Reply:
Yes this line is outdated and even more incorrect at the moment.